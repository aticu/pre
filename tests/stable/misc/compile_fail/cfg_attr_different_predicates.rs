use pre::pre;

#[cfg_attr(target_endian = "big", pre("foo_big"))]
#[cfg_attr(target_endian = "little", pre("foo_little"))]
fn foo() {}

#[pre("bar")]
#[cfg_attr(any(target_endian = "big", target_endian = "little"), pre("baz"))]
fn bar() {}

#[pre]
fn main() {
    #[cfg_attr(target_endian = "big", assure("foo_big", reason = "is foo_big"))]
    #[cfg_attr(
        target_endian = "little",
        assure("foo_little", reason = "is foo_little")
    )]
    foo();

    #[assure("bar", reason = "is bar")]
    #[cfg_attr(
        any(target_endian = "big", target_endian = "little"),
        assure("baz", reason = "is baz")
    )]
    bar();
}
