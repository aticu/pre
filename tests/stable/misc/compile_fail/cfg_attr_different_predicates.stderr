error: mismatched `cfg` predicates for preconditions

  = note: all preconditions must have syntactically equal `cfg` predicates
  = note: `target_endian = "big"` != `target_endian = "little"`

  --> $DIR/cfg_attr_different_predicates.rs:15:9
   |
15 |         target_endian = "little",
   |         ^^^^^^^^^^^^^

error: mismatched `cfg` predicates for preconditions

  = note: all preconditions must have syntactically equal `cfg` predicates
  = note: some preconditions have a `cfg` predicate and some do not

  --> $DIR/cfg_attr_different_predicates.rs:22:9
   |
22 |         any(target_endian = "big", target_endian = "little"),
   |         ^^^

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> $DIR/cfg_attr_different_predicates.rs:18:5
   |
4  |   #[cfg_attr(target_endian = "little", pre("foo_little"))]
   |  __________________________________________-
5  | | fn foo() {}
   | |___________- defined here
...
18 |       foo();
   |       ^^^-- supplied 0 arguments
   |       |
   |       expected 1 argument
